### ISSUE TITLE
Evaluate speech recognition Snowboy Hotword detection

### ISSUE THREAD

**Issue by tdowrick** [UCL/scikit-surgeryspeech#6]
In GitLab by @KimKahl on Jul 23, 2019, 09:13

This API should also be supported by the Speech Recognition API, is offline and free (https://snowboy.kitt.ai/). However, there is no example usage provided from what I've seen now. It might be a bit different to use compared to the other APIs but maybe I will find an example when I search a bit more. @MattClarkson

**Comment by tdowrick** [comment]
In GitLab by @KimKahl on Jul 23, 2019, 09:37

Seems like this is especially for hotword detection and can be used when start listening that the recognition only starts when the keyword is detected. So, I don't thinks it is meant to be the really speech-to-text recognition in the Speech Recognition API. But it might be a good solution to look at if it performs fast, because the time lag between saying start and actually being able to say a command is the biggest problem from my point of view. So I will try out to combine this hotword detection and then the command itself gets recognized by the Google API. Not sure it works, these are just some first thoughts.

**Comment by tdowrick** [comment]
In GitLab by @KimKahl on Jul 23, 2019, 10:34

Unfortunately, I just realized after it didn't work that there is no support for Windows

**Comment by tdowrick** [comment]
In GitLab by @KimKahl on Jul 23, 2019, 10:34

closed

**Comment by tdowrick** [comment]
In GitLab by @KimKahl on Jul 23, 2019, 12:13

reopened

**Comment by tdowrick** [comment]
In GitLab by @KimKahl on Jul 23, 2019, 12:20

Can I use libraries which aren't pip-installable? Because the only other library I found regarding hotword detection was Porcupine (https://github.com/Picovoice/Porcupine) but it's only on github. If I can't use it because of this, further investigation on it might be unnecessary. On the web there are only comparisons between Pocketsphinx which is the one I tried this morning (with the really bad recognition), Snowboy which isn't supported for windows and that one. So if that is not suitable I might just go back to my original solution and try the other API which was for free and see if it has a better performance.
